src/catnap/runtime.rs

```diff
@@ -123,40 +124,45 @@ impl MemoryRuntime for PosixRuntime {
         })
     }
 
-    /// Releases a [dmtr_sgarray_t].
+    /// Releases a scatter-gather array.
     fn free_sgarray(&self, sga: dmtr_sgarray_t) -> Result<(), Fail> {
-        assert_eq!(sga.sga_numsegs, 1);
-        for i in 0..sga.sga_numsegs as usize {
-            let seg: &dmtr_sgaseg_t = &sga.sga_segs[i];
-            let allocation: Box<[u8]> = unsafe {
-                Box::from_raw(slice::from_raw_parts_mut(
-                    seg.sgaseg_buf as *mut _,
-                    seg.sgaseg_len as usize,
-                ))
-            };
-            drop(allocation);
+        // Bad scatter-gather.
```

Comment:
I don't like this declarative comment.  It's like saying "this is a bad scatter-gather", or scatter-gather is bad in general.  Something like "Check for bad scatter-gather array" would be better.

Reply:
Sure. I've changed it to what you suggested.